Add logicmonitor and logicmonitor_facts modules

woz5999 opened the pull request:
##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
logicmonitor / logicmonitor_facts

##### ANSIBLE VERSION
```
ansible 2.1.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### SUMMARY
Due to continued issues and half a day trying to satisfy merge requirements of the travis build, I’m creating another PR which is a continuation of #2442

woz5999 commented:
ready_for_review

woz5999 commented:
Note that this PR is a continuation of previous #2442. shipits have already been received and the only code changes were to docs as requested by jimi-c

gregdek commented:
Thanks @woz5999 for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

gundalow commented on monitoring/logicmonitor.py:
```diff
@@ -0,0 +1,2170 @@
+    description:
+      - The type of LogicMonitor object you wish to manage.
+      - "Collector: Perform actions on a LogicMonitor collector"
+      - NOTE You should use Ansible service modules such as 'service' or 'supervisorctl' for managing the Collector 'logicmonitor-agent' and 'logicmonitor-watchdog' services. Specifically, you'll probably want to start these services after a Collector add and stop these services before a Collector remove.
```
If you use `M(...)` the documentation generator will put in links to those modules

e.g.
`M(service) or M(supervisorctl)`

gundalow commented on monitoring/logicmonitor.py:
```diff
@@ -0,0 +1,2170 @@
+    default: null
+  user:
+    description:
+      - A LogicMonitor user name. The module will authenticate and perform actions on behalf of this user
```
All descriptions should be sentences, need full stops at the end.

gundalow commented on monitoring/logicmonitor_facts.py:
```diff
@@ -0,0 +1,633 @@
+    - LogicMonitor is a hosted, full-stack, infrastructure monitoring platform.
+    - This module collects facts about hosts abd host groups within your LogicMonitor account.
+version_added: "2.2"
+author: Ethan Culler-Mayeno, Jeff Wozniak
```
Can you please change this to Array in the form First Last (@GitHub)

This will make it easier for when we get the Bot to start parsing Modules automatically

gundalow commented:
Few minor comments that apply to all modules.

needs_revision

alikins commented on monitoring/logicmonitor.py:
```diff
@@ -0,0 +1,2170 @@
+            else:
+                return raw
+        except IOError:
+            self.fail(msg="Error: Unknown exception making RPC call")
```
Could be useful to include the errormsg or errno from the caught IOError here.

gregdek commented:
Putting into shipit per @jimi-c in our new modules meeting today. Thanks @woz5999!

jimi-c commented:
Merged, thanks!

alikins commented on monitoring/logicmonitor.py:
```diff
@@ -0,0 +1,2170 @@
+            return f.read()
+        except IOError:
+            # self.module.debug("Error opening URL. " + ioe)
+            self.fail("Unknown exception opening URL")
```
Would like to see the IOError msgs and the url in the message. Though IOError at this point is unlikely to be url specific...

woz5999 commented:
@gundalow @alikins I've got local code changes addresses the comments above. What's the best process for submitting these revisions? Should I push and re-open this PR? Create a new one? What's the preferred workflow for this and going forward.

gundalow commented:
As this product has been merged a fresh PR would be best. 
Thanks
